rpcs3/Emu/Cell/Modules/cellSearch.cpp

```diff
@@ -143,22 +150,120 @@ error_code cellSearchStartContentSearchInList(vm::cptr<CellSearchContentId> list
 
 error_code cellSearchStartContentSearch(CellSearchContentSearchType type, CellSearchSortKey sortKey, CellSearchSortOrder sortOrder, vm::ptr<CellSearchId> outSearchId)
 {
-	cellSearch.todo("cellSearchStartContentSearch(type=0x%x, sortKey=0x%x, sortOrder=0x%x, outSearchId=*0x%x)", (u32) type, (u32) sortKey, (u32) sortOrder, outSearchId);
+	cellSearch.todo("cellSearchStartContentSearch(type=0x%x, sortKey=0x%x, sortOrder=0x%x, outSearchId=*0x%x)", (u32)type, (u32)sortKey, (u32)sortOrder, outSearchId);
 
 	if (!outSearchId)
 	{
 		return CELL_SEARCH_ERROR_PARAM;
 	}
 
+	char* media_dir = (char*)malloc(6);
+	switch (type)
+	{
+		case CELL_SEARCH_CONTENTSEARCHTYPE_MUSIC_ALL: std::strcpy(media_dir, "music"); break;
+		case CELL_SEARCH_CONTENTSEARCHTYPE_PHOTO_ALL: std::strcpy(media_dir, "photo"); break;
+		case CELL_SEARCH_CONTENTSEARCHTYPE_VIDEO_ALL: std::strcpy(media_dir, "video"); break;
+		default: return CELL_SEARCH_ERROR_PARAM;
+	}
+
 	*outSearchId = idm::make<search_object_t>();
 
-	sysutil_register_cb([=](ppu_thread& ppu) -> s32
-	{
+	sysutil_register_cb([=](ppu_thread& ppu) -> s32 {
```

Comment:
{ should always be in the new line

Reply:
I know clangformat is a bit stupid about lambdas

Reply:
This is one of those things I was fighting with Intellisense over... I hadn't yet reviewed all my changed according to the Coding Style page (since this is still a WIP).